pkg/reconciliation/reconcile_racks.go

```diff
@@ -2243,6 +2246,52 @@ func (rc *ReconciliationContext) fixMissingPVC() (bool, error) {
 	return false, nil
 }
 
+func (rc *ReconciliationContext) SetFullQueryLogging() result.ReconcileResult {
+	datacenter := rc.GetDatacenter()
+	var DCConfig map[string]interface{}
+	if err := json.Unmarshal(datacenter.Spec.Config, &DCConfig); err != nil {
+		rc.ReqLogger.Error(err, "error unmarshalling DC config JSON")
+		return result.RequeueSoon(2)
+	}
+	shouldFQLBeEnabled := false
+	if err := DCConfig["full_query_logging_options"]; err == nil {
+		serverMajorVersion, err := strconv.ParseInt(strings.Split(datacenter.Spec.ServerVersion, ".")[0], 10, 8)
+		if err != nil {
+			rc.ReqLogger.Error(err, "error parsing server major version. Can't enable full query logging without knowing this")
+			return result.RequeueSoon(2)
+		}
+		if serverMajorVersion < 4 {
+			err := goerrors.New("full query logging only supported on OSS Cassandra 4x+")
+			rc.ReqLogger.Error(err, "full_query_logging_options is defined in Cassandra config, it is not supported on the version of Cassandra you are running")
+			return result.RequeueSoon(2)
+		}
+		rc.ReqLogger.Info("full_query_logging_options is defined in Cassandra config, we will try to enable it via the management API")
+		shouldFQLBeEnabled = true
```

Comment:
Do we need this parameter even? Can it ever be false and we would continue?

Reply:
Yes, if FQL is currently enabled, but has been removed from the `config.cassandra-yaml` YAML field, then FQL should be turned off at that point.